server/src/main/java/org/opensearch/ingest/IngestService.java

```diff
@@ -1271,6 +1271,14 @@ private static IngestDocumentWrapper toIngestDocumentWrapper(int slot, IndexRequ
         return new IngestDocumentWrapper(slot, toIngestDocument(indexRequest), null);
     }
 
+    private static List<IngestDocumentWrapper> toIngestDocumentWrappers(List<Integer> slots, List<IndexRequest> indexRequests) {
+        List<IngestDocumentWrapper> ingestDocumentWrappers = new ArrayList<>();
+        for (int i = 0; i < slots.size(); ++i) {
+            ingestDocumentWrappers.add(toIngestDocumentWrapper(slots.get(i), indexRequests.get(i)));
+        }
+        return ingestDocumentWrappers;
```

Comment:
nitpick, return immutable list instead:
```java
        return slots.stream()
            .map(i -> toIngestDocumentWrapper(i, indexRequests.get(i)))
            .collect(Collectors.toUnmodifiableList());
```

Reply:
The `i` in `map` is the value in slots not the index which makes `indexRequests.get(i)` incorrect.

Reply:
ah my mistake, you could still make this immutable but feel free to ignore esp bc this is coming from an internal method.